src/brpc/circuit_breaker.cpp

```diff
@@ -188,6 +197,20 @@ bool CircuitBreaker::OnCallEnd(int error_code, int64_t latency) {
     if (_broken.load(butil::memory_order_relaxed)) {
         return false;
     }
+    if (FLAGS_circuit_breaker_half_open_window_size > 0
+        && _half_open.load(butil::memory_order_relaxed)) {
+        if (error_code != 0) {
+            MarkAsBroken();
+            return false;
+        }
+        if (_half_open_success_count.fetch_add(1, butil::memory_order_relaxed)
+                + 1 == FLAGS_circuit_breaker_half_open_window_size) {
+            _half_open.store(false, butil::memory_order_relaxed);
+            _half_open_success_count.store(0, butil::memory_order_relaxed);
+            return true;
```

Comment:
Should we remove the return here and let this latency passed to _long_window/_short_window.OnCallEnd below instead?

Reply:
Done.